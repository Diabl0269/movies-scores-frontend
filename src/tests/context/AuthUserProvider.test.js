const { AuthUserProvider } = require('../../context/AuthUserProvider')
import React from 'react'
import axios from 'axios'
import { shallow } from 'enzyme'

jest.mock('axios')

let wrapper
beforeEach(() => {
  require('../scripts/createLocalStorage')
  wrapper = shallow(<AuthUserProvider />)
})

describe('this is a context for checking auth and getting the user', () => {
  it('should render a loader while getting a user', () => {
    localStorage.setItem('token', 'mockToken')
    const wrapper = shallow(<AuthUserProvider />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })

  xit('should sign up user successfully', async () => {
    const resObject = { data: 1 }
    axios.post.mockReturnValue(resObject)
    const res = wrapper.find('AuthUserProvider').simulate('addUser')
    expect(res).toEqual(resObject)
  })

  xit('should return an object with status 400', async () => {
    const resObject = { status: 400 }
    axios.post.mockReturnValue(resObject)
    const res = await wrapper.addUser()
    expect(res).toEqual(resObject)
  })

  xit('should return an object with status 409', async () => {
    const resObject = { status: 409 }
    axios.post.mockReturnValue(resObject)
    const res = await wrapper.addUser()
    expect(res).toEqual(resObject)
  })
})
