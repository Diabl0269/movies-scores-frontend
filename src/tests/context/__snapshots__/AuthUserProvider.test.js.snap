// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is a context for checking auth and getting the user should render a loader while getting a user 1`] = `ShallowWrapper {}`;

exports[`this is a context for checking auth and getting the user should render successfully 1`] = `ShallowWrapper {}`;
